source/extensions/common/wasm/context.cc

```diff
@@ -1703,6 +1708,65 @@ WasmResult Context::grpcCancel(uint32_t token) {
   return WasmResult::Ok;
 }
 
+WasmResult Context::exprCreate(absl::string_view expr, uint32_t* token_ptr) {
+  auto parse_status = google::api::expr::parser::Parse(std::string(expr));
+  if (!parse_status.ok()) {
+    ENVOY_LOG(info, "wasm expr parse {}: {}", log_prefix(), parse_status.status().message());
+    return WasmResult::BadArgument;
+  }
+
+  // lazily create a builder
+  if (!builder_) {
+    google::api::expr::runtime::InterpreterOptions options;
+    builder_ = google::api::expr::runtime::CreateCelExpressionBuilder(options);
+    auto register_status =
+        google::api::expr::runtime::RegisterBuiltinFunctions(builder_->GetRegistry(), options);
+    if (!register_status.ok()) {
+      return WasmResult::InternalFailure;
+    }
+  }
+
+  google::api::expr::v1alpha1::SourceInfo source_info;
+  auto cel_expression_status =
+      builder_->CreateExpression(&parse_status.ValueOrDie().expr(), &source_info);
+  if (!cel_expression_status.ok()) {
+    ENVOY_LOG(info, "wasm expr compile {}: {}", log_prefix(),
```

Comment:
should this be an error?

Reply:
It's an error in ABI since result is BadArgument. The error message could be copied to the client though, not sure if the client can do anything meaningful with it.